substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -863,6 +863,8 @@ pub mod pallet {
 		ForceEra { mode: Forcing },
 		/// Report of a controller batch deprecation.
 		ControllerBatchDeprecated { failures: u32 },
+		/// Certain amount of bonded fund has been burned.
+		Burnt { stash: T::AccountId, amount: BalanceOf<T> },
```

Comment:
IMO we should just withdraw here, and not burn. Leave the burn (or something else such as transfer these funds to reward pool) logic with the NominationPools.